packages/web-forms/src/demo/FormUpload.vue

```diff
@@ -0,0 +1,334 @@
+<script setup lang="ts">
+import PrimeButton from 'primevue/button';
+import PrimeProgressSpinner from 'primevue/progressspinner';
+import { RouterLink } from 'vue-router';
+
+import PrimeIconField from 'primevue/iconfield';
+import PrimeInputIcon from 'primevue/inputicon';
+import PrimeInputText from 'primevue/inputtext';
+import PrimeMessage from 'primevue/message';
+
+import { computed, ref, watch, type HTMLInputElementEvent } from 'vue';
+import { useConfiguration } from './composables/configuration';
+import { useXlsFormOnline } from './composables/xlsform-online';
+
+const fileInput = ref();
+const highlighted = ref(false);
+const uploadedFilename = ref('');
+
+const xformUrl = ref<string | null>(null);
+
+const error = ref<string>('');
+const warnings = ref<string[]>([]);
+
+const formUploading = ref<boolean>(false);
+
+const { data: config, error: configErrors } = useConfiguration();
+
+const xlsformOnlineBaseUrl = computed(() => config.value?.['xlsform-online-url']);
+
+watch(configErrors, (value) => {
+	if (value) {
+		error.value =
+			'Failed to load configuration, please refresh the page. If problem persists, please report it on ODK Forum.';
+	}
+});
+
+const { convertXlsForm } = useXlsFormOnline(xlsformOnlineBaseUrl);
+
+const fileChangeHandler = async (e: HTMLInputElementEvent) => {
+	if (e.target?.files?.length == 1) {
+		await uploadFile(e.target.files[0]);
+	}
+};
+
+const getFileExtension = (filename: string) => {
+	const parts = filename.split('.');
+	return parts.length > 1 ? parts.pop()! : '';
+};
+
+const fileDropHandler = async (e: DragEvent) => {
+	highlighted.value = false;
+
+	// short-circuit if a file is being uploaded
+	if (formUploading.value) {
+		return;
+	}
+
+	if (!e.dataTransfer || e.dataTransfer.files.length == 0) {
+		return;
+	}
+
+	if (e.dataTransfer.files.length > 1) {
+		error.value = 'You can upload only one file at a time.';
+		return;
+	}
+
+	if (!['xlsx', 'xls', 'xml'].includes(getFileExtension(e.dataTransfer.files[0].name))) {
+		error.value = 'Please upload an Excel file (.xls or .xlsx).';
+		return;
+	}
+
+	await uploadFile(e.dataTransfer.files[0]);
+};
+
+const uploadFile = async (file: File) => {
+	formUploading.value = true;
+	reset();
+	uploadedFilename.value = file.name;
+
+	const { data: response, error: conversionError } = await convertXlsForm(file);
+
+	if (conversionError) {
+		error.value = conversionError;
+	} else if (response) {
+		if (response.error) {
+			error.value = response.error;
+		} else if (response.xform_url) {
+			xformUrl.value = response.xform_url;
+		}
+
+		if (response.warnings && response.warnings.length > 0) {
+			warnings.value = response.warnings;
+		}
+	}
+
+	formUploading.value = false;
+};
+
+const toggleHighlight = () => {
+	// short-circuit if a file is being uploaded
+	if (formUploading.value) {
+		return;
+	}
+
+	highlighted.value = !highlighted.value;
+};
```

Comment:
Nothing for now, but since this is the handler for both `dragenter` and `dragleave`, there are a zillion ways this can produce surprising states. It'll make sense to follow up with:

- distinct enter/leave handlers which set clear states
- (likely depending on need, to be discovered in manual testing) potentially handling for viewport blur, other conditions where `dragenter` state could get "stuck"